staging/src/k8s.io/apiserver/pkg/server/options/encryption_provider_config_test.go

```diff
@@ -0,0 +1,136 @@
+/*
+Copyright 2017 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package options
+
+import (
+	"io/ioutil"
+	"os"
+
+	"k8s.io/apiserver/pkg/storage/value"
+
+	"testing"
+)
+
+var correctConfig string = `
+- kind: k8s-aes-gcm
+  version: v1
+  keys:
+    - name: key1
+      secret: c2VjcmV0IGlzIHNlY3VyZQ==
+    - name: key2
+      secret: dGhpcyBpcyBwYXNzd29yZA==
+  resource: /registry/namespaces
+- kind: k8s-aes-gcm
+  version: v1
+  keys:
+    - name: key2
+      secret: dGhpcyBpcyBwYXNzd29yZA==
+    - name: key3
+      secret: azhzIHNlY3JldCBzdG9yZQ==
+  resource: /registry/
+`
+
+var incorrectConfig1 string = `
+- kind: k8s-aes-gcm
+  version: v1
+  keys:
+    - name: key1
+    - name: key2
+      secret: dGhpcyBpcyBwYXNzd29yZA==
+  resource: /registry/namespaces
+`
+
+var incorrectConfig2 string = `
+- kind: k8s-aes-gcm
+  version: v1
+  keys:
+    - name: key2
+      secret: dGhpcyBpcyBwYXNzd29yZA==
+`
+
+var incorrectConfig3 string = `
+- kind: k8s-aes-gcm
+  version: v1
+  keys:
+    - name: key1
+      secret: YSBzZWNyZXQgYSBzZWNyZXQ=
+  resource: /registry/namespaces
+`
+
+func TestEncryptionProviderConfigCorrectParsing(t *testing.T) {
+	testConfigFile := ".test-enc-provider-correct"
+
+	err := ioutil.WriteFile(testConfigFile, []byte(correctConfig), 0600)
```

Comment:
You could put the parsing code in another function that accepts io.Reader and pass buffers here, avoiding needing to actually write these to disk. Then the top level function just opens the file and passes it to the function that does the actual parsing.